Rewrite introduction and abstract (#477)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

.# Abstract

> This specification defines a mechanism for the selective disclosure of individual elements of a JSON-encoded data structure used as the payload of a JSON Web Signature (JWS).

Reviewer: I'm curious to understand why "JSON-encoded data structure" is used here and elsewhere in the PR vs. "JSON object" or "JSON document" or "bit o' JSON" or something? It reads a bit off to my eye, at first anyway, but I wanted to ask before suggesting changes or otherwise fighting about it.

Author: Mainly to avoid "JSON object" as we can also SD the elements of an array.

Reviewer: Ah ok. I'd read individual elements of a "JSON document" or "bit o' JSON" to mean all the stuff that might be inside that thing like objects or elements of an array. But I take your point or intention.

modified.  However, anyone receiving an unencrypted JWT can read all the
claims. Likewise, anyone with the decryption key receiving encrypted JWT
can also read all the claims.

One of the common use cases of a signed JWT is representing a user's
identity. As long as the signed JWT is one-time
use, it typically only contains those claims the user has consented to
disclose to a specific Verifier. However, there is an increasing number
of use cases where a signed JWT is created once and then used a number
of times by the user (the "Holder" of the JWT). In such use cases, the signed JWT needs
to contain the superset of all claims the user of the
signed JWT might want to disclose to Verifiers at some point. The
ability to selectively disclose a subset of these claims depending on
the Verifier becomes crucial to ensure minimum disclosure and prevent
Verifiers from obtaining claims irrelevant for the transaction at hand.
SD-JWTs defined in this document enable such selective disclosure of JWT claims.

One example of a multi-use JWT is an Issuer-signed
credential that contains the claims about a subject, and whose authenticity can be
cryptographically verified.

> Similar to the JWT specification on which it builds, this document is a product of the

Author: I dropped this part. If we want to keep it, I suggest including it under a separate "## Scope" headline below the intro or below the feature list.

Reviewer: I wrote some of that in response to questions/comments by Hannes about the appropriateness of this work in the OAUTH WG. Maybe the relationship to OAuth and JWT stuff isn't needed anymore. And the generally applicable bit is pretty much implied, in my opinion, even if Dick didn't see it.

> A popular application of JWS are JSON Web Tokens (JWT) [@!RFC7519], which are often used to represent a user's identity.

Author: a```suggestion
A popular application of JWS is JSON Web Token (JWT) [@!RFC7519], a format that is often used to represent a user's identity.
```

> A popular application of JWS are JSON Web Tokens (JWT) [@!RFC7519], which are often used to represent a user's identity.
> An ID Token as defined in OpenID Connect [@?OpenID.Core], for example, is a JWT containing the user's claims created by the server for consumption by a client.

Reviewer 2: ```suggestion
A popular application of JWS are JSON Web Tokens (JWT) [@!RFC7519], which can be used to represent a user's identity.
For example, an ID Token as defined in OpenID Connect [@?OpenID.Core], is a JWT containing the user's identity claims created by a server for consumption by a client.
```
not sure user's identity is the most used application of JWTs..
is it JWT that is an application of JWS, or JWS that is an application of JWT? or they are not applications for either..?
i am not good in articles but why `the` server but `a` client?

> An ID Token as defined in OpenID Connect [@?OpenID.Core], for example, is a JWT containing the user's claims created by the server for consumption by a client.

Reviewer: "client" isn't wrong but has overloaded meaning and isn't used anywhere in this document prior to this PR 
```suggestion
An ID Token as defined in OpenID Connect [@?OpenID.Core], for example, is a JWT containing the user's claims created by the server for consumption by a relying party.
```

> In case the JWS is sent directly from the server to the client, as in OpenID Connect, the server can select which user claims to include in the JWT, minimizing the information shared with the client.

Reviewer 3: Isn't the important fact here that the client is also the party verifying the JWT? I think, adding this detail may be useful to understand the issuer-holder-verifier model defined afterwards as it may otherwise be hard for the reader to understand the fundamental shift.

Reviewer 2: ```suggestion
In case a JWT is sent directly from a server to a client, as in OpenID Connect, the server can select real-time which user claims to include in the JWT, minimizing the information shared with the client, who validates the JWT.
```

Reviewer: reflecting (maybe) prior comments along with my own nitpicks to create this frankenstein of a sentence : 
```suggestion
In cases where the JWT is sent immediately from the server to the relying party, as in OpenID Connect, the server can select at the time of issuance which user claims to include in the JWT, minimizing the information shared with the relying party, who validates the JWT.
```

Reviewer 2: ```suggestion
In cases where the JWT is sent immediately from the server to the relying party, as in OpenID Connect, the server can select at the time of issuance which user claims to include in the JWT, minimizing the information shared with the relying party who validates the JWT.
```

> A new model is emerging that decouples the issuance of a JWT from its presentation.

Reviewer 2: the AS that issues the access token and the RS that consumes it are assumed to be the same entity though, right?

> A new model is emerging that decouples the issuance of a JWT from its presentation.

Reviewer 3: Right. I was thinking in technical terms (server and client communication) while the decoupling happens on a different layer which actually is a new (trust) model. I believe, it would be good to focus more on that change of trust where the RS/Verifier is not a single trusted entity (and certainly not the same entity as the one issuing the JWT) but any third party.

> A new model is emerging that decouples the issuance of a JWT from its presentation.

Reviewer: @ju-cu is not wrong here 
I've long made the complaint that the new "three-party model" sounds silly because we've had 3 party models for a long time.
But not sure how much of that can or should be captured or conveyed here.
And the previous intro might have kind of captured that but not particularly well IMO 
what about a bit o' qualification to this sentence?

```suggestion
A new model is emerging that more fully decouples the issuance of a JWT from its presentation.
```

> A new model is emerging that decouples the issuance of a JWT from its presentation.

Reviewer 2: ```suggestion
A new model is emerging that fully decouples the issuance of a JWT from its presentation.
```

> The Holder would like to disclose only their address to one Verifier, and only their birthdate to a different Verifier.

Reviewer: maybe... i dunno... 
```suggestion
The Holder may elect to disclose only the address to one Verifier, and only the birthdate to a different Verifier.
```

> This model requires a mechanism enabling selective disclosure of data elements while ensuring that Verifiers can still check the authenticity of the data provided.

Reviewer 2: agree that it should be more explicit that selective disclosure becomes an important privacy feature because the same JWT can be used for different purposes.

> This model requires a mechanism enabling selective disclosure of data elements while ensuring that Verifiers can still check the authenticity of the data provided.

Reviewer: something like this?
```suggestion
Privacy principles of minimal disclosure in conjunction with this model demand a mechanism enabling selective disclosure of data elements while ensuring that Verifiers can still check the authenticity of the data provided.
```

> When Key Binding is enforced, a Holder has to proof possession of a private key belonging to a public key contained in the SD-JWT itself.

Reviewer 3: ```suggestion
When Key Binding is enforced, a Holder has to prove possession of a private key belonging to a public key contained in the SD-JWT itself.
```

> It usually does so by signing over a data structure containing transaction-specific data, here defined as the Key Binding JWT.

Reviewer: ```suggestion
It usually does so by signing over a data structure containing transaction-specific data, herein defined as the Key Binding JWT.
```
